subsys/bluetooth/mesh/sensor_cli.c

```diff
@@ -141,17 +142,21 @@ static void handle_status(struct bt_mesh_model *mod,
 
 		sensor_status_id_decode(buf, &length, &id);
 		if (length == 0) {
-			if (is_rsp && rsp->id == id) {
-				rsp->value = NULL;
-				rsp->id = BT_MESH_PROP_ID_PROHIBITED;
-				model_ack_rx(&cli->ack);
+			if (is_rsp && rsp->count == 1) {
+				count = 0;
+				goto yield_ack;
 			}
```

Comment:
We'd still want to send the rest of the sensor values to the callback, even if the requested sensor type wasn't on the server.

I think we should find some middle ground between the old and the new behavior:

```suggestion
			if (is_rsp && rsp->count == 1 && rsp->sensors[0].id == id) {
				is_rsp = false;
				model_ack_rx(&cli->ack);
			}
```

then set `sensors[0].id = BT_MESH_PROP_ID_PROHIBITED;` for the list request, or reinstate the id field of `sensor_data_list_rsp`, and make it `BT_MESH_PROP_ID_PROHIBITED` for list requests to guarantee different behavior between the two cases.